internal/app/lazytable.go

```diff
@@ -10,56 +13,162 @@ import (
 // rowGetter renders the row.
 type rowGetter interface {
 	// Row return a rendered table row.
-	Row(cfg *config.Config) table.Row
+	Row(cfg *config.Config, i int) table.Row
+	Len() int
 }
 
 // lazyTableModel lazily renders table rows.
-type lazyTableModel[T rowGetter] struct {
-	helper
+type lazyTableModel struct {
+	*Application
 
 	table table.Model
 
-	minRenderedRows int
-	allEntries      []T
-	lastCursor      int
+	entries    rowGetter
+	lastCursor int
+	offset     int
+	reverse    bool
+	follow     bool
 
 	renderedRows []table.Row
 }
 
-// Init implements tea.Model.
-func (m lazyTableModel[T]) Init() tea.Cmd {
-	return nil
+type EntriesUpdateMsg struct {
+	Entries rowGetter
 }
 
 // View implements tea.Model.
-func (m lazyTableModel[T]) View() string {
+func (m lazyTableModel) View() string {
 	return m.table.View()
 }
 
 // Update implements tea.Model.
-func (m lazyTableModel[T]) Update(msg tea.Msg) (lazyTableModel[T], tea.Cmd) {
+func (m lazyTableModel) Update(msg tea.Msg) (lazyTableModel, tea.Cmd) {
 	var cmd tea.Cmd
 
+	render := false
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		m, render = m.handleKey(msg, render)
+
+	case EntriesUpdateMsg:
+		m.entries = msg.Entries
+		render = true
+	}
 	m.table, cmd = m.table.Update(msg)
 
 	if m.table.Cursor() != m.lastCursor {
-		m = m.withRenderedRows()
+		render = true
+	}
+
+	if render {
+		m = m.RenderedRows()
 	}
 
 	return m, cmd
 }
 
-func (m lazyTableModel[T]) withRenderedRows() lazyTableModel[T] {
-	cursor := m.table.Cursor()
+func (m lazyTableModel) handleKey(msg tea.KeyMsg, render bool) (lazyTableModel, bool) {
+	if key.Matches(msg, m.Application.keys.Reverse) {
```

Comment:
suggestion: can you please write comments with explanation what's going on here, this function is quite complex to understand?